pages/add-clinic.vue

```diff
@@ -85,39 +75,44 @@
 
 <script>
 import json from "../data/prefectures.json";
+import { required, minLength, url, helpers } from "vuelidate/lib/validators";
+// https://medium.com/@annkilzer/validating-japanese-input-with-vuelidate-and-vue-js-ccdb91d70d36
+const validChars = helpers.regex("validChars", /^[A-Za-z0-9\- .,'"„“’]*$/);
 export default {
-  mounted() {},
   data: () => ({
-    valid: true,
     prefectureList: json.prefectures,
     prefecture: "",
     city: "",
-    cityRules: [
-      (v) => !!v || "City is required",
-      (v) => (v && v.length >= 2) || "City name must be at least 2 characters",
-    ],
     ward: "",
-    wardRules: [
-      (v) => !!v || "Ward is required",
-      (v) => (v && v.length >= 2) || "Ward name must be at least 2 characters",
-    ],
-    name: "",
-    nameRules: [
-      (v) => !!v || "Clinic name is required",
-      (v) =>
-        (v && v.length >= 2) || "Clinic name must be at least 2 characters",
-    ],
+    clinic: "",
     note: "",
     website: "",
-    websiteRules: [
-      (v) => !!v || "Website URL is required",
-      (v) => (v && v.length >= 5) || "Please enter a valid URL",
-    ],
   }),
 
+  validations: {
+    city: {
+      required,
+      minLength: minLength(2),
+      validChars,
+    },
+    ward: {
+      required,
+      minLength: minLength(2),
+      validChars,
+    },
+    clinic: {
+      required,
+      minLength: minLength(2),
+      validChars,
+    },
+    website: {
+      required,
+      url,
+    },
+  },
   methods: {
     submitData() {
-      if (this.$refs.form.validate()) {
+      if (!this.$v.$invalid) {
```

Comment:
is there a `$valid` option vs `!$invalid`?

Reply:
Unfortunately not, that'd make things easier to parse...
https://vuelidate.js.org/#sub-v-values